### PR TITLE
Propagate Api Key UID when it is returned by the service controller.

### PR DESCRIPTION
**Pull request by Elliot-xq**
Propagates the API key UID if it is returned in the service controller CheckResponse.

If it is not returned by the CheckResponse, we will still use the api_key as it is.

### CODE REVIEW

File: src/api_proxy/service_control/request_builder.cc [hunk 1]
```diff
@@ -383,7 +383,9 @@ Status set_credential_id(const SupportedLabel& l, const ReportRequestInfo& info,
     ASSERT(!info.api_key.empty(),
            "API Key must be set, otherwise consumer would not be verified.");
     std::string credential_id("apikey:");
-    credential_id += info.api_key;
+    credential_id = credential_id
+           + (info.check_response_info.api_key_uid.empty() ? info.api_key
+                      : info.check_response_info.api_key_uid);
```
Comment (TAOXUY) [inline src/api_proxy/service_control/request_builder.cc:388]: NIT: define credential_id as constchar above and use absl::StrCat to concat string.

File: src/api_proxy/service_control/request_info.h [hunk 2]
```diff
@@ -176,6 +176,8 @@ struct CheckResponseInfo {
   // The trust level of the API Key that was checked.
   api_key::ApiKeyState api_key_state;
```
Comment (TAOXUY) [inline src/api_proxy/service_control/request_info.h:178]: NIT: add comment

**Review by TAOXUY — APPROVED** [review]

File: src/api_proxy/service_control/request_builder.cc (hunk 1 above)
Comment (Elliot-xq) [reply src/api_proxy/service_control/request_builder.cc:387]: done.

File: src/api_proxy/service_control/request_info.h (hunk 2 above)
Comment (Elliot-xq) [reply src/api_proxy/service_control/request_info.h:178]: done.

File: src/api_proxy/service_control/request_builder.cc (hunk 1 above)
Comment (TAOXUY) [reply src/api_proxy/service_control/request_builder.cc:387]: I mean the constexpr char like in l348

Comment (Elliot-xq) [reply src/api_proxy/service_control/request_builder.cc:388]: done.

**Pull request merged**